Karma/Areas/Identity/Pages/Account/Login.cshtml.cs

```diff
@@ -117,15 +126,19 @@ public async Task<IActionResult> OnPostAsync(string returnUrl = null)
                 if (isLoginWithEmail)
                 {
                     var user = await _userManager.FindByEmailAsync(userName);
-                    if (user == null)
+                    try
                     {
-                        ModelState.AddModelError(string.Empty, "Invalid login attempt.");
-                        return Page();
+                        if (user == null)
+                            throw new InvalidLoginException();
```

Comment:
Is it really necessary to throw exception here if it's going to be caught 4 lines later?

Reply:
If i'm not wrong you have to throw custom exception for it to be cought, might be wrong though.